src/components/Editor/HighlightLine.js

```diff
@@ -59,17 +77,22 @@ export class HighlightLine extends PureComponent<Props> {
     selectedFrame: Frame,
     selectedSource: SourceRecord
   ) {
-    if (!isDocumentReady(selectedSource, selectedLocation)) {
+    const { sourceId, line } = selectedLocation;
+
+    const editorLine = toEditorLine(sourceId, line);
+    if (this.isInSteppingUpdate && editorLine === this.previousLine) {
       return;
     }
+    this.isInSteppingUpdate = false;
+    this.previousLine = editorLine;
 
-    const { sourceId, line } = selectedLocation;
-
+    if (!isDocumentReady(selectedSource, selectedLocation)) {
+      return;
+    }
```

Comment:
some of this logic makes me think we might want to use the lifecycle function `shouldComponentUpdate() {}` to prevent renders...

Reply:
Makes sense, I will look into that.